### ISSUE TITLE
False positive for indentation in a nested lamba

### ISSUE THREAD

**Issue by diklimchuk** [pinterest/ktlint#2624]
## Expected Behavior

No issues

## Observed Behavior

```
Unexpected indentation (8) (should be 12) [Indentation]
Missing newline after "{" [Wrapping]
```

## Steps to Reproduce

I run detekt with ktlint on the following code:
```
@Composable
internal fun tmp(): (Boolean) -> Unit = remember {
    { argument ->
        Unit
    }
}
```

## Your Environment

* Version of ktlint used: detekt "1.23.5"
* Relevant parts of the `.editorconfig` settings: no config
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task):  detekt "1.23.5"
* Version of Gradle used (if applicable): 8.6
* Operating System and version: Mac OS Ventura 13.5 (22G74)

**Comment by paul-dingemans** [comment]
`detekt "1.23.5"` is not a valid ktlint version.

Also, I need more information about how ktlint is configured. With `.editorconfig` below Ktlint `1.1.1` and `1.2.1` both do not reproduce the issue:
```
[*.{kt,kts}]
ktlint_code_style = ktlint_official
ktlint_standard_function-signature = disabled
ktlint_standard_multiline-expression-wrapping = disabled
```

Most likely you can better start with creating an issue in the Detekt project as they know how Detekt integrates with ktlint.

**Comment by paul-dingemans** [comment]
Can not be reproduced. OP has not provided additional information.
